src/httpserver.cpp

```diff
@@ -459,14 +478,28 @@ void StopHTTPServer()
         evhttp_del_accept_socket(eventHTTP, socket);
     }
     boundSockets.clear();
+    {
+        WAIT_LOCK(g_requests_mutex, lock);
+        if (!g_requests.empty()) {
+            LogPrint(BCLog::HTTP, "Waiting for %d requests to stop HTTP server\n", g_requests.size());
+        }
+        while (!g_requests.empty()) {
+            g_requests_cv.wait(lock);
```

Comment:
my suggestion was to add the log here in the loop.

Reply:
Yeah, I think doing it in the loop essentially creates a countdown that I didn't think was necessary. But I am kind of indifferent on this, so I have changed it now.